test_vm/src/lib.rs

```diff
@@ -1,52 +1,599 @@
+            _ => (),
+        }
+        let mut st = self.v.get_state::<InitState>(*INIT_ACTOR_ADDR).unwrap();
+        let target_id = st.map_address_to_new_id(self.v.store, target).unwrap();
+        let target_id_addr = Address::new_id(target_id);
+        let mut init_actor = self.v.get_actor(*INIT_ACTOR_ADDR).unwrap();
+        init_actor.head = self.v.store.put_cbor(&st, Code::Blake2b256).unwrap();
+        self.v.set_actor(*INIT_ACTOR_ADDR, init_actor);
+
+        let new_actor_msg = InternalMessage {
+            from: *SYSTEM_ACTOR_ADDR,
+            to: target_id_addr,
+            value: TokenAmount::zero(),
+            method: METHOD_CONSTRUCTOR,
+            params: serialize::<Address>(target, "address").unwrap(),
+        };
+        {
+            let mut new_ctx = InvocationCtx {
+                v: self.v,
+                top: self.top.clone(),
+                msg: new_actor_msg,
+                allow_side_effects: false,
+                _caller_validated: false,
+                policy: self.policy,
+            };
+            new_ctx.create_actor(*ACCOUNT_ACTOR_CODE_ID, target_id).unwrap();
+            _ = new_ctx.invoke();
+        }
+
+        Ok((self.v.get_actor(target_id_addr).unwrap(), target_id_addr))
+    }
+
+    fn invoke(&mut self) -> Result<RawBytes, ActorError> {
+        let prior_root = self.v.checkpoint();
+        let (mut to_actor, to_addr) = self.resolve_target(&self.msg.to)?;
+
+        // Transfer funds
+        println!("to: {}, from: {}\n", to_addr, self.msg.from);
+        let mut from_actor = self.v.get_actor(self.msg.from).unwrap();
+        if !self.msg.value.is_zero() {
+            if self.msg.value.lt(&BigInt::zero()) {
+                return Err(ActorError::unchecked(
+                    ExitCode::SYS_ASSERTION_FAILED,
+                    "attempt to transfer negative value".to_string(),
+                ));
+            }
+            if from_actor.balance.lt(&self.msg.value) {
+                return Err(ActorError::unchecked(
+                    ExitCode::SYS_INSUFFICIENT_FUNDS,
+                    "insufficient balance to transfer".to_string(),
+                ));
+            }
+        }
+        to_actor.balance = to_actor.balance.add(&self.msg.value);
+        from_actor.balance = from_actor.balance.abs_sub(&self.msg.value);
+        self.v.set_actor(self.msg.from, from_actor);
+        self.v.set_actor(to_addr, to_actor);
+
+        // Exit early on send
+        if self.msg.method == METHOD_SEND {
+            return Ok(RawBytes::default());
+        }
+
+        // call target actor
+        let to_actor = self.v.get_actor(to_addr).unwrap();
+        let params = self.msg.params.clone();
+        let res = match ACTOR_TYPES.get(&to_actor.code).expect("Target actor is not a builtin") {
+            // XXX Review: is there a way to do one call on an object implementing ActorCode trait?
+            // I tried using `dyn` keyword couldn't get the compiler on board.
```

Comment:
@dignifiedquire @Stebalien any ideas would be appreciated 🙏

Reply:
There is not at the moment unfortunately, because `invoke_method` does not take a reference to `self`.